site-manage/datemanager/tool/src/test/java/org/sakaiproject/datemanager/tool/MainControllerCsvTest.java

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (c) 2003-2025 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.datemanager.tool;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReaderBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.fileupload.FileItem;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.datemanager.api.DateManagerService;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.ui.Model;
+import org.springframework.web.servlet.LocaleResolver;
+import org.sakaiproject.util.ResourceLoader;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import org.mockito.MockedStatic;
+import org.mockito.Spy;
+
+/**
+ * Test class for CSV export/import functionality in MainController
+ * Tests both US style (comma) and Spanish/European style (semicolon) CSV formats
+ */
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class MainControllerCsvTest {
+
+    @Mock
+    private DateManagerService dateManagerService;
+    
+    @Mock
+    private SessionManager sessionManager;
+    
+    @Mock
+    private SiteService siteService;
+    
+    @Mock
+    private PreferencesService preferencesService;
+    
+    @Mock
+    private ServerConfigurationService serverConfigurationService;
+    
+    @Mock
+    private HttpServletRequest request;
+    
+    @Mock
+    private HttpServletResponse response;
+    
+    @Mock
+    private Site site;
+    
+    @Mock
+    private LocaleResolver localeResolver;
+    
+    @Mock
+    private Model model;
+    
+    @Spy
+    @InjectMocks
+    private MainController mainController;
```

Comment:
since you have a configuration class these should simply be autowired and the mocks should be in the config class

Reply:
IMO if the csv processing is moved to the service then the tests would also follow and there is already unit tests setup there.